### DISCUSSION TITLE
How to submit OTP for login?

### DISCUSSION THREAD

**Discussion by vinciarts** [dexie/Dexie.js#2027] [Q&A] [answered]
I see the login() method which accepts an email as a parameter, but I couldn’t find any parameters for submitting OTP. Can someone tell me how to submit OTP? Thanks!

**Comment by dfahlander** [comment, marked as answer]
If default authentication is used, the dialog will prompt for email and then otp and send these to server in the backround without having to specify the OTP in the call to db.cloud.login(). But with the latest versions of dexie-cloud-addon, it is possible to pass the otpId and otp along with the email parameter to bypass the dialogs and auto-login user in one step.

Possible parameters:

* email: string
* userId?: string
* grant_type: 'demo' | 'otp'
* otp?: string
* otpId?: string

**Reply by vinciarts** [reply]
Thank you! It now works!
